downloader/file_downloader.go

```diff
@@ -1,34 +1,43 @@
 package downloader
 
 import (
+	"context"
 	"fmt"
-	"io"
-	"net/http"
+	"time"
 
+	"github.com/bitrise-io/go-steputils/input"
+	"github.com/bitrise-io/go-utils/filedownloader"
 	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/retry"
 )
 
 type FileDownloader interface {
-	DownloadFileFromURL(url string) (io.ReadCloser, error)
+	DownloadFileFromURL(url string) ([]byte, error)
 }
 
 type DefaultFileDownloader struct {
-	Logger log.Logger
+	Logger  log.Logger
+	Timeout time.Duration
 }
 
-func (fd *DefaultFileDownloader) DownloadFileFromURL(url string) (io.ReadCloser, error) {
-	resp, err := http.Get(url)
+func (fd *DefaultFileDownloader) DownloadFileFromURL(url string) ([]byte, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), fd.Timeout)
```

Comment:
How does this timeout work? Is this for the whole download operation or just the connection? Becoase in the former case 5 seconds and such won't be enough,

Reply:
What about 30 seconds?

Reply:
What if the artifact is larger? Not sure about the best approach here but for file downloading I'd opt for 2-5 minutes just in case.

Reply:
Okay then, updated